iceoryx_examples/callbacks/ice_callbacks_subscriber.cpp

```diff
@@ -38,7 +38,8 @@ iox::cxx::optional<CounterTopic> rightCache;
 
 static void sigHandler(int f_sig [[gnu::unused]])
 {
-    shutdownSemaphore.post();
+    shutdownSemaphore.post().or_else(
+        [](auto) { std::cerr << "unable to call post on shutdownSemaphore - semaphore corrupt?" << std::endl; });
```

Comment:
in this case a terminate would be necessary, else the application might not stop

Reply:
In the examples I would keep it as simple as possible. Otherwise we also have to create an introduction example for our logger or we have to explain it here somehow but `std::cerr` is understood by everyone.